mlos_bench/db/db_schema.sql

```diff
@@ -0,0 +1,83 @@
+--
+-- Copyright (c) Microsoft Corporation.
+-- Licensed under the MIT License.
+--
+-- DB schema for storing MLOS benchmarking results.
+-- The syntax works for SQLite3, DuckDB, PostgreSQL, and MySQL / MariaDB.
+
+DROP TABLE IF EXISTS trial_telemetry;
+DROP TABLE IF EXISTS trial_results;
+DROP TABLE IF EXISTS trial_config;
+DROP TABLE IF EXISTS experiment_merge;
+DROP TABLE IF EXISTS trial_status;
+DROP TABLE IF EXISTS experiment_config;
+
+CREATE TABLE experiment_config (
+    exp_id VARCHAR(255) NOT NULL,
+    descr TEXT,
+    git_repo VARCHAR(255) NOT NULL,
+    git_commit VARCHAR(40) NOT NULL,
+
+    PRIMARY KEY (exp_id)
+);
+
+CREATE TABLE trial_status (
+    exp_id VARCHAR(255) NOT NULL,
+    trial_id INTEGER NOT NULL,
+    ts_start TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    ts_end TIMESTAMP,
+    -- Should match the text IDs of `mlos_bench.environment.Status` enum:
+    status VARCHAR(16) NOT NULL,
+
+    PRIMARY KEY (exp_id, trial_id),
+    FOREIGN KEY (exp_id) REFERENCES experiment_config(exp_id)
+);
+
+CREATE TABLE experiment_merge (
```

Comment:
Looks like all the supported DBs so far include support for recursive CTEs (`WITH` clauses), so doing the transitive closure to get all trials (both merged and original) for an experiment should be possible with a single query.
Though I'm not sure if sqlalchemy supports that with their ORM and we might end up with some different syntax for different DBs.
Maybe later we can create a `VIEW` in the schema to provide the CTE and then "just" query that with a `SELECT` in the `mlos_bench` code.

Reply:
https://duckdb.org/docs/sql/query_syntax/with.html
https://www.sqlite.org/lang_with.html
https://dev.mysql.com/doc/refman/8.0/en/with.html